### PR TITLE
Proper equivalence for schema types

### PR DESCRIPTION
**Pull request by aaronjeline**
*Description of changes:*
Properly does schema type equivalence given recent changes

### CODE REVIEW

File: cedar-drt/fuzz/src/schemas.rs [hunk 1]
```diff
@@ -100,7 +106,184 @@ fn namespace_equivalence<N: Clone + PartialEq + std::fmt::Debug + std::fmt::Disp
...
+    }
+}
+
+fn entity_type_equivalence<N: Clone + PartialEq + Debug + Display + TypeName>(
+    (name, lhs_type): (UnreservedId, EntityType<N>),
+    rhs: &EntityData<N>,
+) -> Result<(), String> {
+    let rhs_type = rhs
+        .get(&name)
+        .ok_or_else(|| format!("Type `{name}` was missing from right-hand-size"))?;
```
Comment (cdisselkoen) [inline cedar-drt/fuzz/src/schemas.rs:142]:
```suggestion
        .ok_or_else(|| format!("Type `{name}` was missing from right-hand-side"))?;
```

File: cedar-drt/fuzz/src/schemas.rs [hunk 2]
```diff
@@ -100,7 +106,184 @@ fn namespace_equivalence<N: Clone + PartialEq + std::fmt::Debug + std::fmt::Disp
...
+        (
+            TypeVariant::Set {
+                element: lhs_element,
+            },
+            TypeVariant::Set {
+                element: rhs_element,
+            },
+        ) => shape_equiv(lhs_element.as_ref(), rhs_element.as_ref()),
+
+        // Base types are equivalent to `EntityOrCommon` varients where the type_name is of the
```
Comment (cdisselkoen) [inline cedar-drt/fuzz/src/schemas.rs:216]:
```suggestion
        // Base types are equivalent to `EntityOrCommon` variants where the type_name is of the
```

**Review by cdisselkoen — APPROVED** [review]
Looks great, just two typos

File: cedar-drt/fuzz/src/schemas.rs [hunk 3]
```diff
@@ -139,6 +322,18 @@ fn action_type_equivalence<N: PartialEq + std::fmt::Debug + std::fmt::Display>(
...
+/// ApplySpecs are equivalent iff
+/// A) the principal and resource type lists are equal
+/// B) the context shapes are equivalent
+fn apply_spec_equiv<N: TypeName + Clone + PartialEq>(
+    lhs: &ApplySpec<N>,
+    rhs: &ApplySpec<N>,
+) -> bool {
+    shape_equiv(&lhs.context.0, &rhs.context.0)
+        && lhs.principal_types == rhs.principal_types
+        && lhs.resource_types == rhs.resource_types
```
Comment (john-h-kastner-aws) [inline cedar-drt/fuzz/src/schemas.rs:334]: Some point about `Vec` comparison here.

File: cedar-drt/fuzz/src/schemas.rs [hunk 4]
```diff
@@ -100,7 +106,184 @@ fn namespace_equivalence<N: Clone + PartialEq + std::fmt::Debug + std::fmt::Disp
...
+        (Type::Type(lhs), Type::Type(rhs)) => type_var_equiv(lhs, rhs),
+        (Type::CommonTypeRef { type_name: lhs }, Type::CommonTypeRef { type_name: rhs }) => {
+            lhs == rhs
+        }
+        _ => false,
+    }
+}
+
+/// Type Variant equivalence. See the arms of each match for details
+fn type_var_equiv<N: Clone + PartialEq + TypeName>(
```
Comment (john-h-kastner-aws) [inline cedar-drt/fuzz/src/schemas.rs:176]: naming nit: `type_var` reads like `type_variable` to me, so I'd rather spell this out as `type_variant_equiv`

File: cedar-drt/fuzz/src/schemas.rs [hunk 5]
```diff
@@ -100,7 +106,184 @@ fn namespace_equivalence<N: Clone + PartialEq + std::fmt::Debug + std::fmt::Disp
...
+
+fn entity_type_equivalence<N: Clone + PartialEq + Debug + Display + TypeName>(
+    (name, lhs_type): (UnreservedId, EntityType<N>),
+    rhs: &EntityData<N>,
+) -> Result<(), String> {
+    let rhs_type = rhs
+        .get(&name)
+        .ok_or_else(|| format!("Type `{name}` was missing from right-hand-size"))?;
+
+    if lhs_type.member_of_types != rhs_type.member_of_types {
```
Comment (john-h-kastner-aws) [inline cedar-drt/fuzz/src/schemas.rs:144]: I think these are `Vec`s, so this looks like it would error for different orderings

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**